src/System.Slices/src/System/SpanExtensions.cs

```diff
@@ -162,6 +165,69 @@ public static Span<char> Slice(this string str, int start, int length)
             cast[0] = value;
         }
 
+        /// <summary>
+        /// Determines whether two spans are equal by comparing the elements by using memcmp
+        /// </summary>
+        /// <param name="first">A span of type T to compare to second.</param>
+        /// <param name="second">A span of type T to compare to first.</param>
+        public static bool SequenceEqual<T>(this Span<T> first, Span<T> second)
+            where T : struct, IEquatable<T>
+        {
+            if (first.Length != second.Length)
+            {
+                return false;
+            }
+
+            var bytesCount = first.Length * PtrUtils.SizeOf<T>();
+            // pinvoke calls are expensive (mostly due to memory pinning) so we call memcmp only for big slices
+            if (bytesCount >= 512)
+            {
+                return MemoryUtils.MemCmp(first, second);
+            }
```

Comment:
If it is believed to be common enough it might be easier to introduce specialized/optimized overloads like ``` bool SequenceEqual(this Span<int> first, Span<int> second) ``` that could compare in optimized memcmp style.